pages/instructorQuestionManualGrading/instructorQuestionManualGrading.js

```diff
@@ -0,0 +1,104 @@
+                }
+
+                /**
+                 * Student never loaded question (variant and submission is null)
+                 * Student loaded question but did not submit anything (submission is null)
+                 */
+                if (!result.rows[0].variant || !result.rows[0].submission) {
+                    return callback(error.make(404, 'No gradable submissions found.', {locals: res.locals, body: req.body}));
+                }
+
+                res.locals.question = result.rows[0].question;
+                res.locals.variant = result.rows[0].variant;
+                res.locals.submission = result.rows[0].submission;
+                res.locals.score_perc = res.locals.submission.score * 100;
+                callback(null);
+            });
+        },
+       (callback) => {
+            res.locals.overlayGradingInterface = true;
+            question.getAndRenderVariant(res.locals.variant.id, null, res.locals, function (err) {
+              if (ERR(err, next)) return;
+              callback(null);
+            });
+        },
+    ], (err) => {
+        if (ERR(err, next)) return;   
+        res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    });
+
+    debug('GET /');
+});
+
+router.post('/', function(req, res, next) {
+    if (req.body.__action == 'add_manual_grade') {
+        const note = req.body.submission_note;
+        const score = req.body.submission_score;
+        const params = [res.locals.instance_question.id];
+
+        sqlDb.callZeroOrOneRow('instance_question_select_manual_grading_objects', params, (err, result) => {
+            if (ERR(err, next)) return;
+
+            const {question, variant, submission} = result.rows[0];
+            if (!question || !variant || !submission) return next(error.make('500', 'Manual grading dependencies missing'));
+
+            Object.assign(res.locals, {question, variant, submission});
+
+            const params = [
+                submission.id,
+                res.locals.authn_user.user_id,
+                submission.gradable,
+                submission.broken,
+                submission.format_errors,
+                submission.partial_scores,
+                score / 100, // overwrite submission score
+                submission.v2_score,
+                {manual:note}, // overwrite feedback
+                submission.submitted_answer,
+                submission.params,
+                submission.true_answer,
+            ];
+
+            sqlDb.callOneRow('grading_jobs_insert', params, (err, result) => {
+                if (ERR(err, next)) return;
+
+                /* If the submission was marked invalid during grading the grading job will
+                   be marked ungradable and we should bail here to prevent LTI updates. */
+                res.locals['grading_job'] = result.rows[0];
+                if (!res.locals['grading_job'].gradable) return next(error.make('Invalid submission error'));
```

Comment:
```suggestion
                if (!res.locals['grading_job'].gradable) return next(error.make(400, 'Invalid submission error'));
```

Reply:
Actually I'm getting this error when I'm not supposed to...

Reply:
I've changed a query that might make a minor difference here. I am just doing some more QA work to see if I can break stuff. I should hopefully find the reason why you are getting an error when you are not supposed to.

Reply:
I ran into some other minor issues, but I did not find this issue that you were having. It is safe to say that a 400 error code is permissible here regardless.

Reply:
The 400 is required here anyway, that's the first parameter of the make function. As for why I'm getting the error, I'll test it later, maybe it's related to other issues you're fixing?